backend/src/util/VerifyFIN.ts

```diff
@@ -0,0 +1,34 @@
+const checkAlpha = ['X', 'W', 'U', 'T', 'R', 'Q', 'P', 'N', 'M', 'L', 'K'];
+const weightArray = [2, 7, 6, 5, 4, 3, 2];
+
+function validateFIN(fin: string): string {
+  const finUpper = fin.toUpperCase();
```

Comment:
May use `fin.trim().toUpperCase()` in case extra spaces are added.

Reply:
Agree!